.swiftlint.yml

```diff
@@ -364,7 +362,7 @@ line_length: # Lines should not span too many characters.
 
 nesting: # Types should be nested at most 2 level deep, and statements should be nested at most 5 levels deep.
   type_level:
-    warning: 2 # warning - default: 1
+    warning: 4 # warning - default: 1
```

Comment:
This feels like bargaining, but how about 3? 😄

3 would suffice here: https://github.com/Apodini/Apodini/blob/24c8c7dac9e71ba01fe827d31b68fc876e5b10d4/Sources/Apodini/Configurations/Application/Application%2BCore.swift#L32, no?
I have not found 4 levels.

The comment needs adjustment, too.

Reply:
I'm not sure about this but I think if we allow 3 levels and use 3 levels there will be a warning. This is why I went with 4

Reply:
I think in the general case restricting type nesting to max two levels is appropriate, isn’t it? 
Wouldn’t it make more sense to use a ‚//swiftlint:disable[:next]‘ comment if nesting makes sense in your specific case?
I don’t think we want to „allow“ that in general?

again, as this is about code Style, this is a bit of a philosophical discussion, kind of 😅

Reply:
I disagree. There is no reason we shouldn't allow nesting types in general. The main concern behind this rule is that we would end up with too long expressions like this `TypeA.TypeB.TypeC.doStuff()` but this is not how we use nesting of types in general. Our nested types are mostly used inside their parent type and won't result in long expressions. Type nesting helps to keep our code completion clean and helps to categorize/sort and abstract functionality inside our types.

I am fine with disallowing type nesting for `public` types but applying this restriction to `public`, `internal`, `private` and `fileprivate` types is too much collateral damage for my taste.